Runtime/Triangulator.cs

```diff
@@ -603,9 +609,11 @@ private struct DistComparer : IComparer<int>
             public NativeList<int> triangles;
 
             public NativeList<int> halfedges;
-
+            
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<bool> ignorePositions;
             [NativeDisableContainerSafetyRestriction]
-            private NativeArray<int> ids;
+            private NativeList<int> ids;
```

Comment:
You do not have to change array into list.  See the following trick

```csharp
            var a = new NativeArray<int>(64, Allocator.Persistent);
            UnityEngine.Debug.Log(a.Length); // Length = 64

            var count = 0;
            for (int i = 0; i < 32; i++)
            {
                a[count++] = i;
            }

            a = a.GetSubArray(0, count);

            UnityEngine.Debug.Log(a.Length); // Length = 32

            a.Dispose();
```

You can combine `if (ignorePositions.IsCreated && ignorePositions[i])` with this approach to populate the array. Utilizing `Burst` may offer better optimization, although it's not strictly necessary in this case.

Reply:
Neat. Is an array meaningfully faster here though? I thought `NativeList` was just a wrapper around `NativeArray` that does basically this?

Reply:
NativeArray is slightly faster than NativeList because it involves fewer bounds checks, and one less pointer indirection. It can also be easier for LLVM to optimize NativeArrays using SIMD.